pandas/tests/indexes/base_class/test_astype.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+
+from pandas import Index
+
+
+@pytest.mark.parametrize(
```

Comment:
this is not very comprehensive. just hook into the current astype tests and check for this.

Reply:
there are llikely *many* more paths that are not exactly correct.

Reply:
@dsaxton this is the much more important point

Reply:
I'm not sure what's meant by the current astype tests since there are lots of them. Would extending the parameterization work for being more comprehensive?

Reply:
no but i would rather simply add a check i. the current astype tests 

there are many many cases and this doesn’t begin to cover

Reply:
just spitballing, one option would be something like:

```
@pytest.mark.parametrize("to_dtype", [lots, of, stuff])
def test_astype_retains_name(indices, to_dtype):
    indices.name = "foo"  # <-- maybe randomize, include some null-like names
    try:
        result = indices.astype(to_dtype)
    except TypeErrorOrValueErrorOrSomething:
        return
    assert result.name == indices.name
```

Reply:
Not a bad idea, was struggling with how to test this exhaustively when certain conversions won't even be possible

Reply:
@dsaxton did you decide to stick with whats currently here, or have another approach in mind?

Reply:
@jbrockmendel I think I'll try your idea for these tests

Reply:
With this approach I'm assuming we'd skip the MultiIndex cases in the indices fixture, since the name attribute isn't so interesting (we care about the _names_)? Those seem to be all the test failures.

Reply:
that or check that the names are retained.  If this isnt for just the base Index class, then it probably belongs in tests.indexes.test_common